TODO: fix tests for useq-0.9.0

@fdrgsp, we're now getting some test failures (Warnings) with useq v0.9.0 released

https://github.com/pymmcore-plus/pymmcore-widgets/actions/runs/23012555264/job/66826900201?pr=500


```pytb
__________________________ test_sub_wdg_channel_tab ___________________________

qtbot = <pytestqt.qtbot.QtBot object at 0x0000027B42DBB980>
global_mmcore = <CMMCorePlus at 0x27b6825f290 with 16 devices>

    def test_sub_wdg_channel_tab(qtbot: QtBot, global_mmcore: CMMCorePlus) -> None:
        wdg = MDAWidget()
        qtbot.addWidget(wdg)
        wdg.show()
    
        poly1 = useq.GridFromPolygon(
            vertices=[(-400, 0), (1000, -500), (500, 1200), (0, 100)],
            fov_height=100,
            fov_width=100,
            overlap=(10, 10),
        )
    
        seq = useq.MDASequence(
            channels=(
                useq.Channel(config="DAPI", exposure=100),
                useq.Channel(config="FITC", exposure=100),
            ),
            stage_positions=(
>               useq.AbsolutePosition(
                    x=1,
                    y=2,
                    z=3,
                    name="pos1",
                    sequence=useq.MDASequence(grid_plan=poly1),
                ),
            ),
        )

tests\test_useq_core_widgets.py:1073: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = AbsolutePosition(x=1.0, y=2.0, z=3.0, name='pos1', sequence=MDASequence(grid_plan=GridFromPolygon(fov_width=100.0, fov_height=100.0, overlap=(10.0, 10.0), vertices=[(-400.0, 0.0), (1000.0, -500.0), (500.0, 1200.0), (0.0, 100.0)])))

    @model_validator(mode="after")
    def _validate_position(self) -> "Self":
        if self.sequence is None or self.sequence.grid_plan is None:
            return self
        grid = self.sequence.grid_plan
        if not grid.is_relative:
            # x/y are meaningless with an absolute sub-grid (the grid defines
            # them). Warn and clear.
            if self.x is not None or self.y is not None:
>               warnings.warn(
                    f"Position x={self.x!r}, y={self.y!r} is ignored when a position "
                    f"sequence uses an absolute grid plan ({type(grid).__name__}). "
                    "Set x=None, y=None on the position to silence this warning. "
                    "In a future version this will raise an error.",
                    UserWarning,
                    stacklevel=2,
                )
E               UserWarning: Position x=1.0, y=2.0 is ignored when a position sequence uses an absolute grid plan (GridFromPolygon). Set x=None, y=None on the position to silence this warning. In a future version this will raise an error.

.venv\Lib\site-packages\useq\_position.py:117: UserWarning
----------------------------- Captured log setup ------------------------------
```

Do you have time to work on those?
(I'm ok with simply removing any tests that do something silly... but we should also make sure that the widget can't *produce* something that we've now decided is invalid)

--- a/src/pymmcore_widgets/useq_widgets/_mda_sequence.py
+++ b/src/pymmcore_widgets/useq_widgets/_mda_sequence.py
@@ -125,14 +125,26 @@ class MDATabs(CheckableTabWidget):
 
     def value(self) -> useq.MDASequence:
         """Return the current sequence as a [`useq.MDASequence`][]."""
+        grid_plan = self.grid_plan.value() if self.isAxisUsed("g") else None
+        positions = self.stage_positions.value() if self.isAxisUsed("p") else ()
+
+        # If a global absolute grid plan is used, x/y on positions are
+        # meaningless (the grid defines them). Clear them to avoid useq
+        # validation warnings.
+        if grid_plan is not None and not grid_plan.is_relative and positions:
+            positions = tuple(
+                pos.replace(x=None, y=None)
+                if pos.x is not None or pos.y is not None
+                else pos
+                for pos in positions
+            )
+
         return useq.MDASequence(
             z_plan=self.z_plan.value() if self.isAxisUsed("z") else None,
             time_plan=self.time_plan.value() if self.isAxisUsed("t") else None,
-            stage_positions=(
-                self.stage_positions.value() if self.isAxisUsed("p") else ()
-            ),
+            stage_positions=positions,
             channels=self.channels.value() if self.isAxisUsed("c") else (),
-            grid_plan=self.grid_plan.value() if self.isAxisUsed("g") else None,
+            grid_plan=grid_plan,
             metadata={PYMMCW_METADATA_KEY: {"version": pymmcore_widgets.__version__}},
         )
 

--- a/src/pymmcore_widgets/useq_widgets/_positions.py
+++ b/src/pymmcore_widgets/useq_widgets/_positions.py
@@ -245,6 +245,15 @@ class PositionTable(DataTableWidget):
                     # update the sub-sequence dict in the record
                     r["sequence"] = sub_seq
 
+            # If a sub-sequence uses an absolute grid plan, x/y on the
+            # position are meaningless (the grid defines them). Clear them
+            # to avoid useq validation warnings.
+            sub = r.get("sequence")
+            if isinstance(sub, useq.MDASequence) and sub.grid_plan is not None:
+                if not sub.grid_plan.is_relative:
+                    r.pop("x", None)
+                    r.pop("y", None)
+
             pos = useq.Position(**r)
             out.append(pos)
 

--- a/tests/test_useq_core_widgets.py
+++ b/tests/test_useq_core_widgets.py
@@ -1071,8 +1071,6 @@ def test_sub_wdg_channel_tab(qtbot: QtBot, global_mmcore: CMMCorePlus) -> None:
         ),
         stage_positions=(
             useq.AbsolutePosition(
-                x=1,
-                y=2,
                 z=3,
                 name="pos1",
                 sequence=useq.MDASequence(grid_plan=poly1),

